### DISCUSSION TITLE
Typescript SDK export deprecations

### DISCUSSION THREAD

**Discussion by hayes-mysten** [MystenLabs/sui#13150] [General]
This discussion is intended to gather use-cases for deprecated exports in the Typescript SDK

Version `0.38.0` of `@mysten/sui.js` contains a significant refactor that splits the SDK into a set of modular exports.  During this refactor we identified a number of exports which no-longer made sense in the SDK for a variety of reasons, and have been deprecated, and will be entirely removed in a future version of the SDK.

Most exports from `@mysten/sui.js` have moved into a new modular export path like `@mysten/sui.js/client`.  Some exports have been deprecated without replacements, and linked to this discussion.  If you are using any of these deprecated exports that have not been added to a new export, let us know here, and we can figure out how your use case can be supported in future versions of the SDK, or how to create an appropriate replacement

**Comment by retwn** [comment]
How it will look like to use the functions `getExecutionStatusError`, `getExecutionStatusType`, `getObjectFields`, in future versions of the SDK?

**Reply by hayes-mysten** [reply]
`getExecutionStatusError(data)` -> `data?.effects?.status.error`
`getExecutionStatusType(data)` -> `transaction?.effects?.status.status`
`getObjectFields(data)` -> `data?.content?.dataType === 'moveObject' ? data?.content.fields : null`

`getObjectFields` is the only one of these that isn't just a simple lookup in the response, because the content on an object can be either an object or a package.  We could potentially add a helper for this back into `@mysten/sui.js/utils` if there is demand for it.  The abstractions in the existing getters hide complexity in ways that can lead to more confusion than they solve, so if we do add these back in we want to replace them with new versions that are a little easier to reason about.
